### PR TITLE
fix makefile bug: not defined RM, so just use rm

### PR DESCRIPTION
**Pull request by charlie-wong**
git ignore also add "CMakeLists.txt.user" for Qt Creator

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,7 +28,7 @@ checkprefix:
 	  cached_prefix=$(shell $(CMAKE_PRG) -L -N build | 2>/dev/null grep 'CMAKE_INSTALL_PREFIX' | cut -d '=' -f2); \
 	  if ! [ "$(CMAKE_INSTALL_PREFIX)" = "$$cached_prefix" ]; then \
 	    printf "Re-running CMake: CMAKE_INSTALL_PREFIX '$(CMAKE_INSTALL_PREFIX)' does not match cached value '%s'.\n" "$$cached_prefix"; \
-	    $(RM) build/.ran-cmake; \
```
Comment (jamessan) [inline Makefile:31]: `RM` is supposed to be a standard variable defined by make.  What version of make are you using?  What does `make -p` show?

**Comment by dundargoc** [comment]
What is the purpose of using `RM`? Yes I read the documentation, and no it didn't say anything useful.

**Comment by jamessan** [comment]
> What is the purpose of using `RM`?

Like any abstraction, it means that the value can be changed in one place and all uses are updated.

> I couldn't find any info on it.

https://www.gnu.org/software/make/manual/make.html#index-RM

**Comment by dundargoc** [comment]
> gnu.org/software/make/manual/make.html#index-RM

I know :). I guess I should've said I couldn't find any documentation that explained why it was useful.

> Like any abstraction, it means that the value can be changed in one place and all uses are updated.

Gotcha, thanks for the explanation.

**Comment by dundargoc** [comment]
The author is not responding and this doesn't seem to affect anyone else. Closing until one of these changes.

**Pull request closed without merging**